src/EnergyPlus/HeatBalanceManager.cc

```diff
@@ -4515,12 +4516,140 @@ namespace HeatBalanceManager {
 			} // GroupNum
 		}
 
+		GetZoneLocalEnvData( ErrorsFound );
+
 		//allocate the array the holds the predefined report data
 		ZonePreDefRep.allocate( NumOfZones );
 
 	}
 
 	void
+	GetZoneLocalEnvData( bool & ErrorsFound ) // Error flag indicator (true if errors found)
+	{
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR         X LUO
+		//       DATE WRITTEN   July 2017
+		//       MODIFIED       na
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS SUBROUTINE:
+		// load input data for Outdoor Air Node for zones
+
+		// METHODOLOGY EMPLOYED:
+		// usual E+ input processes
+
+
+		// Using/Aliasing
+		using namespace DataIPShortCuts;
+		using InputProcessor::GetNumObjectsFound;
+		using InputProcessor::GetObjectItem;
+		using InputProcessor::GetObjectDefMaxArgs;
+		using InputProcessor::GetNumObjectsFound;
+		using InputProcessor::FindItemInList;
+		using InputProcessor::VerifyName;
+
+		using NodeInputManager::GetOnlySingleNode;
+		using OutAirNodeManager::CheckOutAirNodeNumber;
+
+		using DataHeatBalance::ZoneLocalEnvironment;
+		using DataLoopNode::NodeType_Air;
+		using DataLoopNode::NodeConnectionType_Inlet;
+		using DataLoopNode::ObjectIsParent;
+
+		// Locals
+		// SUBROUTINE ARGUMENT DEFINITIONS:
+
+		// SUBROUTINE PARAMETER DEFINITIONS:
+		static std::string const RoutineName( "GetZoneLocalEnvData: " );
+
+		// INTERFACE BLOCK SPECIFICATIONS:na
+		// DERIVED TYPE DEFINITIONS:na
+		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+		int NumArgs;
+		int NumAlpha;
+		int NumNumeric;
+
+		int Loop;
+		int ZoneLoop;
+		int ZoneNum; // DO loop counter for zones
+		int TotZoneEnv;
+		bool ErrorInName;
+		bool IsBlank;
+		int IOStat;
+		int NodeNum;
+
+		//-----------------------------------------------------------------------
+		//               ZoneProperty:LocalEnvironment
+		//-----------------------------------------------------------------------
+
+		cCurrentModuleObject = "ZoneProperty:LocalEnvironment";		
+		TotZoneEnv = GetNumObjectsFound( cCurrentModuleObject );
+		GetObjectDefMaxArgs( cCurrentModuleObject, NumArgs, NumAlpha, NumNumeric );
+		if ( NumAlpha != 3 ) {
+			ShowSevereError( RoutineName + cCurrentModuleObject + ": Object Definition indicates wroung number of Alpha Objects (requires 3)." );
```

Comment:
Fixed. This is getting from a SurfaceProperty:ExteriorNaturalVentedCavity reading file I copied to learn the structure. void GetHTSurfExtVentedCavityData( bool & ErrorsFound ) to check IDD definication. But that's property old ways. Changed the SurfaceProperty reading as well.